src/corona/models/ring.jl

```diff
@@ -283,26 +400,26 @@ function _ring_arm!(
     _, min_ρ_index = findmin(ρs)
     _, max_ρ_index = findmax(ρs)
 
-    min_ρ_index, max_ρ_index = min(min_ρ_index, max_ρ_index), max(min_ρ_index, max_ρ_index)
+    r1, r2 = _split_arms_indices(angles, ρs)
 
     # TODO: do we need views here? should go through this and work out where the memory gets copied
     left = @views Gradus._process_ring_traces(
         cache.setup,
         m,
         d,
         cache.v,
-        vcat(gps[1:(min_ρ_index-1)], gps[max_ρ_index+1:end]),
-        vcat(ρs[1:(min_ρ_index-1)], ρs[max_ρ_index+1:end]),
-        vcat(angles[1:(min_ρ_index-1)], angles[max_ρ_index+1:end]),
+        gps[r1],
+        ρs[r1],
+        angles[r1],
     )
     right = @views Gradus._process_ring_traces(
         cache.setup,
         m,
         d,
         cache.v,
-        gps[min_ρ_index:max_ρ_index],
-        ρs[min_ρ_index:max_ρ_index],
-        angles[min_ρ_index:max_ρ_index],
+        gps[r2],
+        ρs[r2],
+        angles[r2],
     )
```

Comment:
The surrounding function here can be removed, since this code assumes only two branches.